tests/constants1.py

```diff
@@ -19,15 +19,15 @@ def test_ord_chr():
 
 
 def test_abs():
-    # a: i32
-    # a = abs(5)
-    # a = abs(-500)
-    # a = abs(False)
-    # a = abs(True)
+    a: i32
+    a = abs(5)
```

Comment:
I think this should create the correct "value" (at compile time). However, I think it also creates a `FunctionCall` that calls `abs(float)` with `abs(5)` and that will fail verify (once we update verify to be more strict).

What we need is to first fix #141, and then implement a runtime version of `abs` for other types than float.

Reply:
I think this is still producing an incorrect ASR, see the reference results. So it should not be committed in my opinion.

Reply:
The new reference ASR in this PR contains:
```
(FunctionCall 4 abs () [(ConstantInteger 5 (Integer 4 []))] [] (Real 8 []) (ConstantInteger 5 (Integer 4 [])) ()) ()) 
```
Which is incorrect, because it is calling an abs function that expects a float, but passing in integer as an argument. So we should not commit this test, as this is not correct.

I'll try to robustify our verify very soon, so that it triggers an error, then it will be obvious.